Active version details returned by service/:id/details

ovv opened the pull request:
Some information is not made accessible from the `service/:id/details` request. This is useful to reduce the number of overall API calls made to fastly.

This change is not perfectly backward compatible as `ServiceDetail.ActiveVersion` changes type. If this is not acceptable we can figure out an alternative.

Integralist commented:
Hi @ovv 👋🏻 

Thanks for opening this pull request and your patience waiting for a reply (I appreciate this was opened a while ago). 

I've been working on some changes to Go-Fastly recently that _might_ warrant a bump in major version and I stumbled across this PR as a potential candidate for being included in that new release as it is a breaking change.

If that's the case and the major version bump is something we want to do then would you be interested in helping get this PR included in that release? It would require rebasing the latest master and ensuring there are appropriate tests in place to validate the changes in this PR.

If you could let me know. Thanks!

ovv commented:
Hello @Integralist 

Yeah it would be great to have this merged. I'll take a look next week.

ovv commented:
Hello @Integralist , I rebased and added a simple test. I don't have a fastly key on hand to update the fixture so the test is failing.

Last time I tested this code was working fine, it would be great if you (or someone else from fastly) could update the fixture. Thanks.

Integralist commented:
Hi @ovv I pulled your branch locally and modified the fixture files by setting up a test service and passing in the values when running the tests. It looked something like:

```sh
# remove the 'services' fixture files
rm -r fastly/fixtures/services/*

# run specific set of tests that will cause the fixture files to be recreated
# I provide a real service id + use a real api token associated with that service
make test FASTLY_TEST_SERVICE_ID="..." FASTLY_API_KEY="..." TESTARGS="-run=TestClient_Services"

# replace my real service id with a default one used by all other fixtures
make fix-fixtures FASTLY_TEST_SERVICE_ID="..."
```

Below are the changes I needed to make to get the tests to pass (you'll notice I had to tweak the code slightly because it seemed that embedding the `Version` struct inside of `ActiveVersion` wasn't allowing the `ActiveVersion.Number` to unmarshal correctly so I inlined those fields from the `Version` struct instead. There's probably a way to solve that (and if you're aware of how to do that then great) but it didn't seem necessary to go down a rabbit hole for the sake of a few extra lines of code.

```diff
diff --git a/fastly/fixtures/services/cleanup.yaml b/fastly/fixtures/services/cleanup.yaml
index 7e3c843..cf79210 100644
--- a/fastly/fixtures/services/cleanup.yaml
+++ b/fastly/fixtures/services/cleanup.yaml
@@ -10,7 +10,7 @@ interactions:
       - xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
       User-Agent:
       - FastlyGo/0.4.3.dev (+github.com/fastly/go-fastly; go1.12)
-    url: https://api.fastly.com/service/4cyy8qad5xykKfjyTCkGnq
+    url: https://api.fastly.com/service/7i6HN3TK9wS159v2gPAZ8A
     method: DELETE
   response:
     body: '{"status":"ok"}'
@@ -54,7 +54,7 @@ interactions:
       - xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
       User-Agent:
       - FastlyGo/0.4.3.dev (+github.com/fastly/go-fastly; go1.12)
-    url: https://api.fastly.com/service/4cyy8qad5xykKfjyTCkGnq
+    url: https://api.fastly.com/service/7i6HN3TK9wS159v2gPAZ8A
     method: DELETE
   response:
     body: '{"status":"ok"}'
diff --git a/fastly/fixtures/services/create.yaml b/fastly/fixtures/services/create.yaml
index 66b6bdb..b811971 100644
--- a/fastly/fixtures/services/create.yaml
+++ b/fastly/fixtures/services/create.yaml
@@ -19,7 +19,7 @@ interactions:
     url: https://api.fastly.com/service
     method: POST
   response:
-    body: '{"customer_id":"76eo17QE8I25lqXy9uJqbd","comment":"comment","name":"test-service","created_at":"2019-05-11T17:03:47Z","publish_key":"ff100f0c03c61271b4052b07878159c721c4e897","deleted_at":null,"versions":[{"staging":false,"updated_at":"2019-05-11T17:03:47Z","deployed":false,"comment":"","service_id":"4cyy8qad5xykKfjyTCkGnq","locked":false,"number":1,"deleted_at":null,"testing":false,"created_at":"2019-05-11T17:03:47Z","active":false}],"id":"4cyy8qad5xykKfjyTCkGnq","updated_at":"2019-05-11T17:03:47Z"}'
+    body: '{"customer_id":"76eo17QE8I25lqXy9uJqbd","comment":"comment","name":"test-service","created_at":"2019-05-11T17:03:47Z","publish_key":"ff100f0c03c61271b4052b07878159c721c4e897","deleted_at":null,"versions":[{"staging":false,"updated_at":"2019-05-11T17:03:47Z","deployed":false,"comment":"","service_id":"7i6HN3TK9wS159v2gPAZ8A","locked":false,"number":1,"deleted_at":null,"testing":false,"created_at":"2019-05-11T17:03:47Z","active":false}],"id":"7i6HN3TK9wS159v2gPAZ8A","updated_at":"2019-05-11T17:03:47Z"}'
     headers:
       Accept-Ranges:
       - bytes
diff --git a/fastly/fixtures/services/delete.yaml b/fastly/fixtures/services/delete.yaml
index 808e77f..7798476 100644
--- a/fastly/fixtures/services/delete.yaml
+++ b/fastly/fixtures/services/delete.yaml
@@ -10,7 +10,7 @@ interactions:
       - xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
       User-Agent:
       - FastlyGo/0.4.3.dev (+github.com/fastly/go-fastly; go1.12)
-    url: https://api.fastly.com/service/4cyy8qad5xykKfjyTCkGnq
+    url: https://api.fastly.com/service/7i6HN3TK9wS159v2gPAZ8A
     method: DELETE
   response:
     body: '{"status":"ok"}'
diff --git a/fastly/fixtures/services/details.yaml b/fastly/fixtures/services/details.yaml
index d04680a..45da6fb 100644
--- a/fastly/fixtures/services/details.yaml
+++ b/fastly/fixtures/services/details.yaml
@@ -1,30 +1,28 @@
 ---
 version: 1
-rwmutex: {}
 interactions:
 - request:
     body: ""
     form: {}
     headers:
-      Fastly-Key:
-      - xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
       User-Agent:
-      - FastlyGo/0.4.3.dev (+github.com/fastly/go-fastly; go1.12)
-    url: https://api.fastly.com/service/4cyy8qad5xykKfjyTCkGnq/details
+      - FastlyGo/2.1.0 (+github.com/fastly/go-fastly; go1.15.4)
+    url: https://api.fastly.com/service/7i6HN3TK9wS159v2gPAZ8A/details
     method: GET
   response:
-    body: '{"name":"test-service","created_at":"2019-05-11T17:03:47Z","updated_at":"2019-05-11T17:03:47Z","deleted_at":null,"versions":[{"created_at":"2019-05-11T17:03:47Z","locked":false,"active":false,"staging":false,"deleted_at":null,"testing":false,"comment":"","service_id":"4cyy8qad5xykKfjyTCkGnq","number":1,"deployed":false,"updated_at":"2019-05-11T17:03:47Z"}],"id":"4cyy8qad5xykKfjyTCkGnq","comment":"comment","customer_id":"76eo17QE8I25lqXy9uJqbd","version":{"service_id":"4cyy8qad5xykKfjyTCkGnq","staging":false,"updated_at":"2019-05-11T17:03:47Z","created_at":"2019-05-11T17:03:47Z","locked":false,"number":1,"deployed":false,"active":false,"deleted_at":null,"comment":"","testing":false,"acls":[],"backends":[],"cache_settings":[],"conditions":[],"dictionaries":[],"directors":[],"domains":[],"gzips":[],"headers":[],"healthchecks":[],"request_settings":[],"response_objects":[],"snippets":[],"vcls":[],"wordpress":[],"settings":{"general.default_host":"","general.default_ttl":3600,"general.stale_if_error":false,"general.stale_if_error_ttl":43200,"general.default_pci":0}},"active_version":null}'
+    body: '{"created_at":"2021-01-13T18:56:56Z","type":"vcl","versions":[{"created_at":"2021-01-13T18:56:56Z","testing":false,"deployed":false,"deleted_at":null,"staging":false,"comment":"","active":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","updated_at":"2021-01-13T18:56:56Z","number":1,"locked":false},{"active":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","deleted_at":null,"comment":"","staging":false,"deployed":false,"testing":false,"created_at":"2021-01-13T19:00:27Z","locked":false,"number":2,"updated_at":"2021-01-13T19:00:31Z"},{"updated_at":"2021-01-13T19:00:35Z","number":3,"locked":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","testing":false,"created_at":"2021-01-13T19:00:32Z","active":false,"deployed":false,"comment":"","deleted_at":null,"staging":false},{"number":4,"updated_at":"2021-01-13T19:00:34Z","service_id":"7i6HN3TK9wS159v2gPAZ8A","locked":false,"created_at":"2021-01-13T19:00:33Z","testing":false,"active":false,"staging":false,"deleted_at":null,"comment":"","deployed":false},{"number":5,"updated_at":"2021-01-13T19:00:36Z","service_id":"7i6HN3TK9wS159v2gPAZ8A","locked":false,"testing":false,"created_at":"2021-01-13T19:00:33Z","deleted_at":null,"comment":"","staging":false,"deployed":false,"active":false},{"locked":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","number":6,"updated_at":"2021-01-13T19:00:37Z","active":false,"deleted_at":null,"staging":false,"comment":"","deployed":false,"created_at":"2021-01-13T19:00:34Z","testing":false},{"service_id":"7i6HN3TK9wS159v2gPAZ8A","locked":false,"number":7,"updated_at":"2021-01-13T19:00:39Z","comment":"","staging":false,"deleted_at":null,"deployed":false,"active":false,"testing":false,"created_at":"2021-01-13T19:00:34Z"},{"created_at":"2021-01-13T19:00:34Z","testing":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","active":false,"deployed":false,"deleted_at":null,"comment":"","staging":false,"updated_at":"2021-01-13T19:00:37Z","number":8,"locked":false},{"locked":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","updated_at":"2021-01-13T19:00:39Z","number":9,"deployed":false,"deleted_at":null,"comment":"","staging":false,"active":false,"created_at":"2021-01-13T19:00:35Z","testing":false},{"created_at":"2021-01-13T19:00:35Z","testing":false,"deployed":false,"deleted_at":null,"staging":false,"comment":"","active":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","updated_at":"2021-01-13T19:00:38Z","number":10,"locked":false},{"testing":false,"created_at":"2021-01-13T19:00:36Z","service_id":"7i6HN3TK9wS159v2gPAZ8A","active":false,"deleted_at":null,"comment":"","staging":false,"deployed":false,"number":11,"updated_at":"2021-01-13T19:00:38Z","locked":false},{"created_at":"2021-01-13T19:00:36Z","testing":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","active":false,"staging":false,"comment":"","deleted_at":null,"deployed":false,"number":12,"updated_at":"2021-01-13T19:00:39Z","locked":false},{"testing":false,"created_at":"2021-01-13T19:00:37Z","service_id":"7i6HN3TK9wS159v2gPAZ8A","active":false,"deployed":false,"comment":"","deleted_at":null,"staging":false,"updated_at":"2021-01-13T19:00:39Z","number":13,"locked":false},{"number":14,"updated_at":"2021-01-13T19:00:39Z","locked":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","testing":false,"created_at":"2021-01-13T19:00:37Z","deleted_at":null,"comment":"","staging":false,"deployed":false,"active":false},{"locked":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","number":15,"updated_at":"2021-01-13T19:00:38Z","active":false,"comment":"","staging":false,"deleted_at":null,"deployed":false,"created_at":"2021-01-13T19:00:37Z","testing":false},{"number":16,"updated_at":"2021-01-13T19:00:38Z","service_id":"7i6HN3TK9wS159v2gPAZ8A","locked":false,"created_at":"2021-01-13T19:00:38Z","testing":false,"staging":false,"deleted_at":null,"comment":"","deployed":false,"active":false},{"deployed":false,"comment":"new
+      comment","staging":false,"deleted_at":null,"active":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","created_at":"2021-01-14T09:51:27Z","testing":false,"locked":true,"updated_at":"2021-01-14T09:51:30Z","number":17},{"locked":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","updated_at":"2021-01-14T09:51:30Z","number":18,"deployed":false,"comment":"","staging":false,"deleted_at":null,"active":false,"testing":false,"created_at":"2021-01-14T09:51:30Z"},{"deployed":false,"staging":false,"deleted_at":null,"comment":"","service_id":"7i6HN3TK9wS159v2gPAZ8A","active":false,"testing":false,"created_at":"2021-01-14T10:25:24Z","locked":false,"updated_at":"2021-01-14T10:25:27Z","number":19},{"number":20,"updated_at":"2021-01-14T10:25:52Z","service_id":"7i6HN3TK9wS159v2gPAZ8A","locked":false,"testing":false,"created_at":"2021-01-14T10:25:50Z","active":false,"staging":false,"comment":"","deleted_at":null,"deployed":false},{"active":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","comment":"","staging":false,"deleted_at":null,"deployed":false,"testing":false,"created_at":"2021-01-14T10:25:50Z","locked":false,"number":21,"updated_at":"2021-01-14T10:25:51Z"},{"deployed":false,"deleted_at":null,"staging":false,"comment":"","service_id":"7i6HN3TK9wS159v2gPAZ8A","active":false,"testing":false,"created_at":"2021-01-14T10:26:20Z","locked":false,"updated_at":"2021-01-14T10:26:23Z","number":22},{"number":23,"updated_at":"2021-01-14T10:26:25Z","locked":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","testing":false,"created_at":"2021-01-14T10:26:23Z","active":false,"deleted_at":null,"staging":false,"comment":"","deployed":false},{"service_id":"7i6HN3TK9wS159v2gPAZ8A","active":false,"deployed":false,"deleted_at":null,"comment":"","staging":false,"created_at":"2021-01-14T10:33:35Z","testing":false,"locked":false,"updated_at":"2021-01-14T10:33:38Z","number":24},{"active":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","deployed":false,"staging":false,"deleted_at":null,"comment":"","testing":false,"created_at":"2021-01-14T10:33:53Z","locked":false,"updated_at":"2021-01-14T10:33:55Z","number":25},{"updated_at":"2021-01-14T10:35:02Z","number":26,"locked":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","testing":false,"created_at":"2021-01-14T10:35:00Z","deployed":false,"comment":"","deleted_at":null,"staging":false,"active":false},{"updated_at":"2021-01-14T10:35:44Z","number":27,"locked":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","created_at":"2021-01-14T10:35:41Z","testing":false,"active":false,"deployed":false,"staging":false,"comment":"","deleted_at":null},{"service_id":"7i6HN3TK9wS159v2gPAZ8A","active":false,"comment":"","deleted_at":null,"staging":false,"deployed":false,"created_at":"2021-01-14T10:35:48Z","testing":false,"locked":false,"number":28,"updated_at":"2021-01-14T10:35:55Z"},{"testing":false,"created_at":"2021-01-14T10:36:00Z","deleted_at":null,"comment":"","staging":false,"deployed":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","active":false,"number":29,"updated_at":"2021-01-14T10:36:01Z","locked":false},{"created_at":"2021-01-14T10:36:13Z","testing":false,"deployed":false,"deleted_at":null,"comment":"","staging":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","active":false,"updated_at":"2021-01-14T10:36:16Z","number":30,"locked":false},{"updated_at":"2021-01-14T10:36:25Z","number":31,"locked":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","created_at":"2021-01-14T10:36:23Z","testing":false,"active":false,"deployed":false,"comment":"","staging":false,"deleted_at":null},{"testing":false,"created_at":"2021-01-14T10:40:44Z","service_id":"7i6HN3TK9wS159v2gPAZ8A","active":false,"comment":"","deleted_at":null,"staging":false,"deployed":false,"number":32,"updated_at":"2021-01-14T10:40:47Z","locked":false},{"created_at":"2021-01-14T10:40:53Z","testing":false,"active":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","deployed":false,"deleted_at":null,"staging":false,"comment":"","updated_at":"2021-01-14T10:40:54Z","number":33,"locked":false},{"created_at":"2021-01-14T10:41:16Z","testing":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","active":false,"deleted_at":null,"staging":false,"comment":"","deployed":false,"number":34,"updated_at":"2021-01-14T10:41:16Z","locked":false},{"service_id":"7i6HN3TK9wS159v2gPAZ8A","locked":false,"number":35,"updated_at":"2021-01-14T10:41:17Z","staging":false,"deleted_at":null,"comment":"","deployed":false,"active":false,"created_at":"2021-01-14T10:41:16Z","testing":false},{"testing":false,"created_at":"2021-01-14T10:41:23Z","deleted_at":null,"staging":false,"comment":"","deployed":false,"active":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","number":36,"updated_at":"2021-01-14T10:41:25Z","locked":false},{"service_id":"7i6HN3TK9wS159v2gPAZ8A","active":false,"comment":"","staging":false,"deleted_at":null,"deployed":false,"testing":false,"created_at":"2021-01-14T10:41:32Z","locked":false,"number":37,"updated_at":"2021-01-14T10:41:35Z"},{"number":38,"updated_at":"2021-01-14T10:41:42Z","service_id":"7i6HN3TK9wS159v2gPAZ8A","locked":false,"testing":false,"created_at":"2021-01-14T10:41:40Z","active":false,"staging":false,"deleted_at":null,"comment":"","deployed":false},{"deleted_at":null,"staging":false,"comment":"","deployed":false,"active":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","testing":false,"created_at":"2021-01-14T10:41:49Z","locked":false,"number":39,"updated_at":"2021-01-14T10:41:51Z"},{"locked":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","number":40,"updated_at":"2021-01-14T10:42:24Z","active":false,"comment":"","staging":false,"deleted_at":null,"deployed":false,"testing":false,"created_at":"2021-01-14T10:42:17Z"},{"active":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","comment":"","deleted_at":null,"staging":false,"deployed":false,"testing":false,"created_at":"2021-01-14T10:42:18Z","locked":false,"number":41,"updated_at":"2021-01-14T10:42:24Z"},{"active":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","deployed":false,"deleted_at":null,"staging":false,"comment":"","created_at":"2021-01-14T10:42:29Z","testing":false,"locked":false,"updated_at":"2021-01-14T10:42:35Z","number":42},{"service_id":"7i6HN3TK9wS159v2gPAZ8A","active":false,"comment":"","staging":false,"deleted_at":null,"deployed":false,"testing":false,"created_at":"2021-01-14T10:42:40Z","locked":false,"number":43,"updated_at":"2021-01-14T10:42:43Z"},{"updated_at":"2021-01-14T10:42:49Z","number":44,"service_id":"7i6HN3TK9wS159v2gPAZ8A","locked":false,"testing":false,"created_at":"2021-01-14T10:42:47Z","deployed":false,"deleted_at":null,"comment":"","staging":false,"active":false},{"updated_at":"2021-01-14T10:43:02Z","number":45,"service_id":"7i6HN3TK9wS159v2gPAZ8A","locked":false,"created_at":"2021-01-14T10:43:00Z","testing":false,"active":false,"deployed":false,"deleted_at":null,"staging":false,"comment":""},{"updated_at":"2021-01-14T10:43:08Z","number":46,"locked":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","created_at":"2021-01-14T10:43:06Z","testing":false,"active":false,"deployed":false,"comment":"","deleted_at":null,"staging":false},{"deleted_at":null,"staging":false,"comment":"","deployed":false,"active":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","testing":false,"created_at":"2021-01-14T10:43:13Z","locked":false,"number":47,"updated_at":"2021-01-14T10:43:15Z"},{"testing":false,"created_at":"2021-01-14T10:43:29Z","active":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","deleted_at":null,"staging":false,"comment":"","deployed":false,"number":48,"updated_at":"2021-01-14T10:43:29Z","locked":false},{"service_id":"7i6HN3TK9wS159v2gPAZ8A","locked":false,"number":49,"updated_at":"2021-01-14T10:44:05Z","active":false,"deleted_at":null,"staging":false,"comment":"","deployed":false,"created_at":"2021-01-14T10:44:03Z","testing":false},{"service_id":"7i6HN3TK9wS159v2gPAZ8A","locked":false,"updated_at":"2021-01-14T10:44:11Z","number":50,"active":false,"deployed":false,"staging":false,"comment":"","deleted_at":null,"created_at":"2021-01-14T10:44:09Z","testing":false},{"testing":false,"created_at":"2021-01-14T10:44:16Z","deployed":false,"comment":"","staging":false,"deleted_at":null,"service_id":"7i6HN3TK9wS159v2gPAZ8A","active":false,"updated_at":"2021-01-14T10:44:18Z","number":51,"locked":false},{"number":52,"updated_at":"2021-01-14T10:44:29Z","service_id":"7i6HN3TK9wS159v2gPAZ8A","locked":false,"testing":false,"created_at":"2021-01-14T10:44:27Z","active":false,"deleted_at":null,"staging":false,"comment":"","deployed":false},{"created_at":"2021-01-14T10:45:47Z","testing":false,"active":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","deployed":false,"deleted_at":null,"staging":false,"comment":"","updated_at":"2021-01-14T10:45:50Z","number":53,"locked":false},{"active":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","staging":false,"deleted_at":null,"comment":"","deployed":false,"testing":false,"created_at":"2021-01-14T10:46:11Z","locked":false,"number":54,"updated_at":"2021-01-14T10:46:12Z"},{"testing":false,"created_at":"2021-01-14T10:46:16Z","service_id":"7i6HN3TK9wS159v2gPAZ8A","active":false,"deployed":false,"staging":false,"deleted_at":null,"comment":"","updated_at":"2021-01-14T10:46:19Z","number":55,"locked":false},{"number":56,"updated_at":"2021-01-14T10:46:46Z","locked":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","created_at":"2021-01-14T10:46:44Z","testing":false,"staging":false,"comment":"","deleted_at":null,"deployed":false,"active":false},{"updated_at":"2021-01-14T10:46:52Z","number":57,"service_id":"7i6HN3TK9wS159v2gPAZ8A","locked":false,"created_at":"2021-01-14T10:46:50Z","testing":false,"deployed":false,"staging":false,"comment":"","deleted_at":null,"active":false},{"number":58,"updated_at":"2021-01-14T10:47:04Z","locked":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","testing":false,"created_at":"2021-01-14T10:46:58Z","active":false,"comment":"","staging":false,"deleted_at":null,"deployed":false},{"number":59,"updated_at":"2021-01-14T10:47:11Z","service_id":"7i6HN3TK9wS159v2gPAZ8A","locked":false,"created_at":"2021-01-14T10:47:09Z","testing":false,"deleted_at":null,"staging":false,"comment":"","deployed":false,"active":false},{"number":60,"updated_at":"2021-01-14T10:47:30Z","service_id":"7i6HN3TK9wS159v2gPAZ8A","locked":false,"created_at":"2021-01-14T10:47:30Z","testing":false,"comment":"","staging":false,"deleted_at":null,"deployed":false,"active":false},{"updated_at":"2021-01-14T10:47:44Z","number":61,"service_id":"7i6HN3TK9wS159v2gPAZ8A","locked":false,"created_at":"2021-01-14T10:47:42Z","testing":false,"deployed":false,"comment":"","staging":false,"deleted_at":null,"active":false},{"deployed":false,"deleted_at":null,"comment":"","staging":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","active":false,"created_at":"2021-01-14T10:47:53Z","testing":false,"locked":false,"updated_at":"2021-01-14T10:47:56Z","number":62},{"number":63,"updated_at":"2021-01-14T11:03:06Z","locked":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","created_at":"2021-01-14T11:02:59Z","testing":false,"active":false,"deleted_at":null,"staging":false,"comment":"","deployed":false},{"number":64,"updated_at":"2021-01-14T11:05:45Z","locked":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","testing":false,"created_at":"2021-01-14T11:05:43Z","active":false,"staging":false,"deleted_at":null,"comment":"","deployed":false},{"deleted_at":null,"comment":"","staging":false,"deployed":false,"active":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","created_at":"2021-01-14T11:08:32Z","testing":false,"locked":false,"number":65,"updated_at":"2021-01-14T11:08:35Z"},{"number":66,"updated_at":"2021-01-14T11:10:04Z","service_id":"7i6HN3TK9wS159v2gPAZ8A","locked":false,"created_at":"2021-01-14T11:10:02Z","testing":false,"deleted_at":null,"comment":"","staging":false,"deployed":false,"active":false},{"locked":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","updated_at":"2021-01-14T12:03:24Z","number":67,"active":false,"deployed":false,"comment":"","deleted_at":null,"staging":false,"created_at":"2021-01-14T12:03:24Z","testing":false},{"service_id":"7i6HN3TK9wS159v2gPAZ8A","locked":false,"updated_at":"2021-01-14T12:04:14Z","number":68,"deployed":false,"deleted_at":null,"comment":"","staging":false,"active":false,"created_at":"2021-01-14T12:04:12Z","testing":false},{"locked":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","updated_at":"2021-01-14T12:05:12Z","number":69,"deployed":false,"staging":false,"comment":"","deleted_at":null,"active":false,"created_at":"2021-01-14T12:05:09Z","testing":false},{"service_id":"7i6HN3TK9wS159v2gPAZ8A","locked":true,"number":70,"updated_at":"2021-01-14T14:36:25Z","staging":false,"comment":"","deleted_at":null,"deployed":false,"active":false,"testing":false,"created_at":"2021-01-14T12:06:21Z"},{"deleted_at":null,"staging":false,"comment":"","deployed":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","active":true,"created_at":"2021-01-14T14:22:42Z","testing":false,"locked":true,"number":71,"updated_at":"2021-01-14T14:36:25Z"}],"name":"test-service","id":"7i6HN3TK9wS159v2gPAZ8A","deleted_at":null,"customer_id":"51MumwLiSJyFTWhtbByYgR","comment":"comment","updated_at":"2021-01-14T14:22:25Z","version":{"locked":true,"created_at":"2021-01-14T14:22:42Z","active":true,"updated_at":"2021-01-14T14:36:25Z","testing":false,"deleted_at":null,"deployed":false,"number":71,"comment":"","staging":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","acls":[],"backends":[{"ssl_ca_cert":null,"use_ssl":true,"ipv6":null,"weight":100,"override_host":null,"max_conn":200,"address":"httpbin.org","name":"Host
+      1","ssl_sni_hostname":"httpbin.org","ssl_cert_hostname":"httpbin.org","between_bytes_timeout":10000,"hostname":"httpbin.org","ssl_client_key":null,"ssl_check_cert":true,"port":443,"ipv4":null,"comment":"","ssl_client_cert":null,"healthcheck":null,"ssl_hostname":null,"client_cert":null,"first_byte_timeout":15000,"ssl_ciphers":null,"request_condition":"","auto_loadbalance":false,"min_tls_version":null,"shield":null,"max_tls_version":null,"error_threshold":0,"connect_timeout":1000}],"cache_settings":[],"conditions":[],"dictionaries":[],"directors":[],"domains":[{"name":"integralist-test-go-fastly-tests.com","comment":"comment"}],"gzips":[],"headers":[],"healthchecks":[],"request_settings":[],"response_objects":[],"snippets":[],"vcls":[],"wordpress":[],"settings":{"general.stale_if_error":false,"general.default_pci":0,"general.default_host":"","general.default_ttl":3600,"general.stale_if_error_ttl":43200}},"active_version":{"locked":true,"created_at":"2021-01-14T14:22:42Z","active":true,"updated_at":"2021-01-14T14:36:25Z","testing":false,"deleted_at":null,"deployed":false,"number":71,"comment":"","staging":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","acls":[],"backends":[{"ssl_ca_cert":null,"use_ssl":true,"ipv6":null,"weight":100,"override_host":null,"max_conn":200,"address":"httpbin.org","name":"Host
+      1","ssl_sni_hostname":"httpbin.org","ssl_cert_hostname":"httpbin.org","between_bytes_timeout":10000,"hostname":"httpbin.org","ssl_client_key":null,"ssl_check_cert":true,"port":443,"ipv4":null,"comment":"","ssl_client_cert":null,"healthcheck":null,"ssl_hostname":null,"client_cert":null,"first_byte_timeout":15000,"ssl_ciphers":null,"request_condition":"","auto_loadbalance":false,"min_tls_version":null,"shield":null,"max_tls_version":null,"error_threshold":0,"connect_timeout":1000}],"cache_settings":[],"conditions":[],"dictionaries":[],"directors":[],"domains":[{"name":"integralist-test-go-fastly-tests.com","comment":"comment"}],"gzips":[],"headers":[],"healthchecks":[],"request_settings":[],"response_objects":[],"snippets":[],"vcls":[],"wordpress":[],"settings":{"general.stale_if_error":false,"general.default_pci":0,"general.default_host":"","general.default_ttl":3600,"general.stale_if_error_ttl":43200}}}'
     headers:
       Accept-Ranges:
       - bytes
-      Age:
-      - "0"
       Cache-Control:
       - no-cache
       Content-Type:
       - application/json
       Date:
-      - Sat, 11 May 2019 17:03:48 GMT
+      - Thu, 14 Jan 2021 14:36:48 GMT
       Status:
       - 200 OK
       Strict-Transport-Security:
@@ -32,15 +30,15 @@ interactions:
       Vary:
       - Accept-Encoding
       Via:
-      - 1.1 varnish
-      - 1.1 varnish
+      - 1.1 varnish, 1.1 varnish
       X-Cache:
       - MISS, MISS
       X-Cache-Hits:
       - 0, 0
       X-Served-By:
-      - cache-control-slwdc9037-CONTROL-SLWDC, cache-bur17542-BUR
+      - cache-control-slwdc9035-CONTROL-SLWDC, cache-man4129-MAN
       X-Timer:
-      - S1557594228.476050,VS0,VE395
+      - S1610635009.690048,VS0,VE268
     status: 200 OK
     code: 200
+    duration: ""
diff --git a/fastly/fixtures/services/domain.yaml b/fastly/fixtures/services/domain.yaml
index 2732115..9f2709b 100644
--- a/fastly/fixtures/services/domain.yaml
+++ b/fastly/fixtures/services/domain.yaml
@@ -10,7 +10,7 @@ interactions:
       - xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
       User-Agent:
       - FastlyGo/0.4.3.dev (+github.com/fastly/go-fastly; go1.12)
-    url: https://api.fastly.com/service/4cyy8qad5xykKfjyTCkGnq/domain
+    url: https://api.fastly.com/service/7i6HN3TK9wS159v2gPAZ8A/domain
     method: GET
   response:
     body: '[]'
diff --git a/fastly/fixtures/services/get.yaml b/fastly/fixtures/services/get.yaml
index 3f9800e..27bbe4e 100644
--- a/fastly/fixtures/services/get.yaml
+++ b/fastly/fixtures/services/get.yaml
@@ -10,10 +10,10 @@ interactions:
       - xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
       User-Agent:
       - FastlyGo/0.4.3.dev (+github.com/fastly/go-fastly; go1.12)
-    url: https://api.fastly.com/service/4cyy8qad5xykKfjyTCkGnq
+    url: https://api.fastly.com/service/7i6HN3TK9wS159v2gPAZ8A
     method: GET
   response:
-    body: '{"name":"test-service","customer_id":"76eo17QE8I25lqXy9uJqbd","publish_key":"ff100f0c03c61271b4052b07878159c721c4e897","created_at":"2019-05-11T17:03:47Z","comment":"comment","updated_at":"2019-05-11T17:03:47Z","id":"4cyy8qad5xykKfjyTCkGnq","versions":[{"created_at":"2019-05-11T17:03:47Z","locked":false,"active":false,"staging":false,"deleted_at":null,"testing":false,"comment":"","service_id":"4cyy8qad5xykKfjyTCkGnq","number":1,"deployed":false,"updated_at":"2019-05-11T17:03:47Z"}],"deleted_at":null}'
+    body: '{"name":"test-service","customer_id":"76eo17QE8I25lqXy9uJqbd","publish_key":"ff100f0c03c61271b4052b07878159c721c4e897","created_at":"2019-05-11T17:03:47Z","comment":"comment","updated_at":"2019-05-11T17:03:47Z","id":"7i6HN3TK9wS159v2gPAZ8A","versions":[{"created_at":"2019-05-11T17:03:47Z","locked":false,"active":false,"staging":false,"deleted_at":null,"testing":false,"comment":"","service_id":"7i6HN3TK9wS159v2gPAZ8A","number":1,"deployed":false,"updated_at":"2019-05-11T17:03:47Z"}],"deleted_at":null}'
     headers:
       Accept-Ranges:
       - bytes
diff --git a/fastly/fixtures/services/list.yaml b/fastly/fixtures/services/list.yaml
index 3c71539..7ccae31 100644
--- a/fastly/fixtures/services/list.yaml
+++ b/fastly/fixtures/services/list.yaml
@@ -15,7 +15,7 @@ interactions:
   response:
     body: '[{"customer_id":"76eo17QE8I25lqXy9uJqbd","comment":"Managed by Terraform","versions":[{"active":true,"deployed":false,"number":1,"locked":true,"comment":"","testing":false,"staging":false,"service_id":"5ITAdHmjs0z86y7j2WlLqU","deleted_at":null,"updated_at":"2019-05-02T18:42:07Z","created_at":"2019-05-02T18:40:31Z"}],"id":"5ITAdHmjs0z86y7j2WlLqU","version":1,"name":"demofastly","updated_at":"2019-05-02T18:40:31Z","created_at":"2019-05-02T18:40:31Z"},{"version":1,"name":"test","updated_at":"2019-04-17T00:15:03Z","created_at":"2019-04-17T00:15:03Z","comment":"Managed
       by Terraform","id":"7i6HN3TK9wS159v2gPAZ8A","versions":[{"testing":false,"deployed":false,"active":true,"comment":"","number":1,"locked":true,"updated_at":"2019-04-17T00:15:10Z","created_at":"2019-04-17T00:15:03Z","service_id":"7i6HN3TK9wS159v2gPAZ8A","staging":false,"deleted_at":null},{"created_at":"2019-04-17T23:35:33Z","updated_at":"2019-04-17T23:36:00Z","deleted_at":null,"staging":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","testing":false,"locked":false,"number":2,"comment":"","active":false,"deployed":false},{"testing":false,"deployed":false,"active":false,"number":3,"locked":false,"comment":"","created_at":"2019-04-19T18:19:12Z","updated_at":"2019-05-10T02:49:35Z","staging":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","deleted_at":null},{"created_at":"2019-05-07T21:34:43Z","updated_at":"2019-05-07T21:34:43Z","deleted_at":null,"service_id":"7i6HN3TK9wS159v2gPAZ8A","staging":false,"testing":false,"comment":"","number":4,"locked":false,"deployed":false,"active":false},{"testing":false,"comment":"","locked":false,"number":5,"deployed":false,"active":false,"updated_at":"2019-05-07T21:45:40Z","created_at":"2019-05-07T21:45:40Z","deleted_at":null,"service_id":"7i6HN3TK9wS159v2gPAZ8A","staging":false},{"testing":false,"deployed":false,"active":false,"comment":"","number":6,"locked":false,"updated_at":"2019-05-07T21:51:04Z","created_at":"2019-05-07T21:51:02Z","service_id":"7i6HN3TK9wS159v2gPAZ8A","staging":false,"deleted_at":null},{"testing":false,"number":7,"locked":false,"comment":"","active":false,"deployed":false,"created_at":"2019-05-07T22:38:34Z","updated_at":"2019-05-07T22:38:36Z","deleted_at":null,"staging":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A"},{"created_at":"2019-05-07T23:54:37Z","updated_at":"2019-05-07T23:54:40Z","deleted_at":null,"staging":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","testing":false,"number":8,"locked":false,"comment":"","active":false,"deployed":false},{"testing":false,"active":false,"deployed":false,"number":9,"locked":false,"comment":"","updated_at":"2019-05-08T03:58:02Z","created_at":"2019-05-08T03:58:00Z","staging":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","deleted_at":null},{"created_at":"2019-05-08T05:27:27Z","updated_at":"2019-05-08T05:27:29Z","staging":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","deleted_at":null,"testing":false,"deployed":false,"active":false,"number":10,"locked":false,"comment":""},{"deployed":false,"active":false,"locked":false,"number":11,"comment":"","testing":false,"staging":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","deleted_at":null,"updated_at":"2019-05-08T18:03:10Z","created_at":"2019-05-08T18:03:08Z"},{"staging":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","deleted_at":null,"updated_at":"2019-05-08T18:24:09Z","created_at":"2019-05-08T18:24:07Z","deployed":false,"active":false,"number":12,"locked":false,"comment":"","testing":false},{"testing":false,"number":13,"locked":false,"comment":"","deployed":false,"active":false,"created_at":"2019-05-08T18:42:11Z","updated_at":"2019-05-08T18:42:13Z","deleted_at":null,"staging":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A"},{"testing":false,"comment":"","locked":false,"number":14,"deployed":false,"active":false,"created_at":"2019-05-08T21:53:06Z","updated_at":"2019-05-08T21:53:08Z","deleted_at":null,"service_id":"7i6HN3TK9wS159v2gPAZ8A","staging":false},{"testing":false,"comment":"","number":15,"locked":false,"deployed":false,"active":false,"updated_at":"2019-05-08T22:01:06Z","created_at":"2019-05-08T22:01:01Z","deleted_at":null,"service_id":"7i6HN3TK9wS159v2gPAZ8A","staging":false},{"testing":false,"comment":"","number":16,"locked":false,"active":false,"deployed":false,"updated_at":"2019-05-08T22:38:28Z","created_at":"2019-05-08T22:38:27Z","deleted_at":null,"service_id":"7i6HN3TK9wS159v2gPAZ8A","staging":false},{"testing":false,"deployed":false,"active":false,"comment":"","locked":false,"number":17,"updated_at":"2019-05-08T22:55:02Z","created_at":"2019-05-08T22:54:59Z","service_id":"7i6HN3TK9wS159v2gPAZ8A","staging":false,"deleted_at":null},{"comment":"","number":18,"locked":false,"deployed":false,"active":false,"testing":false,"deleted_at":null,"service_id":"7i6HN3TK9wS159v2gPAZ8A","staging":false,"updated_at":"2019-05-08T23:18:11Z","created_at":"2019-05-08T23:18:08Z"},{"testing":false,"deployed":false,"active":false,"locked":false,"number":19,"comment":"","created_at":"2019-05-08T23:35:12Z","updated_at":"2019-05-08T23:35:14Z","staging":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","deleted_at":null},{"testing":false,"active":false,"deployed":false,"number":20,"locked":false,"comment":"","created_at":"2019-05-08T23:59:03Z","updated_at":"2019-05-08T23:59:03Z","staging":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","deleted_at":null},{"service_id":"7i6HN3TK9wS159v2gPAZ8A","staging":false,"deleted_at":null,"created_at":"2019-05-09T00:00:13Z","updated_at":"2019-05-09T00:00:17Z","active":false,"deployed":false,"comment":"","number":21,"locked":false,"testing":false},{"created_at":"2019-05-09T17:06:21Z","updated_at":"2019-05-09T17:06:21Z","staging":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","deleted_at":null,"testing":false,"deployed":false,"active":false,"locked":false,"number":22,"comment":""},{"created_at":"2019-05-09T17:19:18Z","updated_at":"2019-05-09T17:19:20Z","deleted_at":null,"service_id":"7i6HN3TK9wS159v2gPAZ8A","staging":false,"testing":false,"comment":"","locked":false,"number":23,"deployed":false,"active":false},{"testing":false,"deployed":false,"active":false,"locked":false,"number":24,"comment":"","updated_at":"2019-05-09T17:19:58Z","created_at":"2019-05-09T17:19:56Z","staging":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","deleted_at":null},{"testing":false,"locked":false,"number":25,"comment":"","deployed":false,"active":false,"created_at":"2019-05-09T18:07:23Z","updated_at":"2019-05-09T18:07:27Z","deleted_at":null,"staging":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A"},{"testing":false,"comment":"","number":26,"locked":false,"active":false,"deployed":false,"updated_at":"2019-05-09T19:53:10Z","created_at":"2019-05-09T19:53:10Z","deleted_at":null,"service_id":"7i6HN3TK9wS159v2gPAZ8A","staging":false},{"deployed":false,"active":false,"comment":"","locked":false,"number":27,"testing":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","staging":false,"deleted_at":null,"updated_at":"2019-05-09T19:53:13Z","created_at":"2019-05-09T19:53:10Z"},{"comment":"","locked":false,"number":28,"active":false,"deployed":false,"testing":false,"deleted_at":null,"service_id":"7i6HN3TK9wS159v2gPAZ8A","staging":false,"updated_at":"2019-05-09T20:01:14Z","created_at":"2019-05-09T20:01:12Z"},{"testing":false,"deployed":false,"active":false,"comment":"","locked":false,"number":29,"created_at":"2019-05-09T20:06:50Z","updated_at":"2019-05-09T20:06:53Z","service_id":"7i6HN3TK9wS159v2gPAZ8A","staging":false,"deleted_at":null},{"testing":false,"deployed":false,"active":false,"comment":"","number":30,"locked":false,"created_at":"2019-05-09T20:18:21Z","updated_at":"2019-05-09T20:18:21Z","service_id":"7i6HN3TK9wS159v2gPAZ8A","staging":false,"deleted_at":null},{"created_at":"2019-05-09T20:19:44Z","updated_at":"2019-05-09T20:19:44Z","deleted_at":null,"service_id":"7i6HN3TK9wS159v2gPAZ8A","staging":false,"testing":false,"comment":"","number":31,"locked":false,"deployed":false,"active":false},{"created_at":"2019-05-09T20:20:33Z","updated_at":"2019-05-09T20:20:35Z","deleted_at":null,"staging":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","testing":false,"number":32,"locked":false,"comment":"","active":false,"deployed":false},{"deleted_at":null,"staging":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","updated_at":"2019-05-09T20:26:17Z","created_at":"2019-05-09T20:26:14Z","number":33,"locked":false,"comment":"","deployed":false,"active":false,"testing":false},{"deleted_at":null,"service_id":"7i6HN3TK9wS159v2gPAZ8A","staging":false,"created_at":"2019-05-09T20:34:11Z","updated_at":"2019-05-09T20:34:15Z","comment":"","number":34,"locked":false,"deployed":false,"active":false,"testing":false},{"deleted_at":null,"staging":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","updated_at":"2019-05-09T21:58:06Z","created_at":"2019-05-09T21:58:03Z","number":35,"locked":false,"comment":"","deployed":false,"active":false,"testing":false},{"service_id":"7i6HN3TK9wS159v2gPAZ8A","staging":false,"deleted_at":null,"updated_at":"2019-05-09T22:07:02Z","created_at":"2019-05-09T22:07:00Z","active":false,"deployed":false,"comment":"","locked":false,"number":36,"testing":false},{"updated_at":"2019-05-09T22:16:14Z","created_at":"2019-05-09T22:16:14Z","deleted_at":null,"service_id":"7i6HN3TK9wS159v2gPAZ8A","staging":false,"testing":false,"comment":"","number":37,"locked":false,"active":false,"deployed":false},{"comment":"","number":38,"locked":false,"active":false,"deployed":false,"testing":false,"deleted_at":null,"service_id":"7i6HN3TK9wS159v2gPAZ8A","staging":false,"created_at":"2019-05-09T22:19:39Z","updated_at":"2019-05-09T22:19:39Z"},{"number":39,"locked":false,"comment":"","active":false,"deployed":false,"testing":false,"deleted_at":null,"staging":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","created_at":"2019-05-09T22:22:36Z","updated_at":"2019-05-09T22:22:36Z"},{"testing":false,"comment":"","number":40,"locked":false,"deployed":false,"active":false,"updated_at":"2019-05-09T22:29:25Z","created_at":"2019-05-09T22:29:23Z","deleted_at":null,"service_id":"7i6HN3TK9wS159v2gPAZ8A","staging":false},{"updated_at":"2019-05-09T22:30:59Z","created_at":"2019-05-09T22:30:56Z","deleted_at":null,"staging":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","testing":false,"number":41,"locked":false,"comment":"","active":false,"deployed":false},{"testing":false,"deployed":false,"active":false,"comment":"","number":42,"locked":false,"updated_at":"2019-05-09T22:50:18Z","created_at":"2019-05-09T22:50:15Z","service_id":"7i6HN3TK9wS159v2gPAZ8A","staging":false,"deleted_at":null},{"testing":false,"number":43,"locked":false,"comment":"","deployed":false,"active":false,"updated_at":"2019-05-09T23:08:45Z","created_at":"2019-05-09T23:08:45Z","deleted_at":null,"staging":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A"},{"active":false,"deployed":false,"comment":"","locked":false,"number":44,"testing":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","staging":false,"deleted_at":null,"created_at":"2019-05-09T23:13:36Z","updated_at":"2019-05-09T23:13:36Z"},{"service_id":"7i6HN3TK9wS159v2gPAZ8A","staging":false,"deleted_at":null,"updated_at":"2019-05-09T23:21:13Z","created_at":"2019-05-09T23:21:13Z","active":false,"deployed":false,"comment":"","locked":false,"number":45,"testing":false},{"updated_at":"2019-05-09T23:22:00Z","created_at":"2019-05-09T23:21:57Z","service_id":"7i6HN3TK9wS159v2gPAZ8A","staging":false,"deleted_at":null,"testing":false,"active":false,"deployed":false,"comment":"","locked":false,"number":46},{"locked":false,"number":47,"comment":"","active":false,"deployed":false,"testing":false,"deleted_at":null,"staging":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","updated_at":"2019-05-09T23:38:59Z","created_at":"2019-05-09T23:38:56Z"},{"comment":"","locked":false,"number":48,"active":false,"deployed":false,"testing":false,"deleted_at":null,"service_id":"7i6HN3TK9wS159v2gPAZ8A","staging":false,"updated_at":"2019-05-10T18:17:58Z","created_at":"2019-05-10T18:17:55Z"},{"updated_at":"2019-05-11T00:21:36Z","created_at":"2019-05-11T00:21:34Z","deleted_at":null,"staging":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","testing":false,"number":49,"locked":true,"comment":"new
-      comment","active":false,"deployed":false},{"deleted_at":null,"service_id":"7i6HN3TK9wS159v2gPAZ8A","staging":false,"updated_at":"2019-05-11T00:21:36Z","created_at":"2019-05-11T00:21:36Z","comment":"","number":50,"locked":false,"active":false,"deployed":false,"testing":false},{"testing":false,"active":false,"deployed":false,"comment":"","locked":false,"number":51,"updated_at":"2019-05-11T00:32:55Z","created_at":"2019-05-11T00:32:53Z","service_id":"7i6HN3TK9wS159v2gPAZ8A","staging":false,"deleted_at":null}],"customer_id":"76eo17QE8I25lqXy9uJqbd"},{"comment":"comment","id":"4cyy8qad5xykKfjyTCkGnq","versions":[{"created_at":"2019-05-11T17:03:47Z","updated_at":"2019-05-11T17:03:47Z","deleted_at":null,"staging":false,"service_id":"4cyy8qad5xykKfjyTCkGnq","testing":false,"number":1,"locked":false,"comment":"","active":false,"deployed":false}],"customer_id":"76eo17QE8I25lqXy9uJqbd","created_at":"2019-05-11T17:03:47Z","updated_at":"2019-05-11T17:03:47Z","name":"test-service","version":1},{"id":"5stIxTZWQz7D8B1Am1Mghm","versions":[{"updated_at":"2019-05-03T16:02:42Z","created_at":"2019-05-03T16:02:37Z","deleted_at":null,"staging":false,"service_id":"5stIxTZWQz7D8B1Am1Mghm","testing":false,"locked":true,"number":1,"comment":"","active":true,"deployed":false},{"deleted_at":null,"staging":false,"service_id":"5stIxTZWQz7D8B1Am1Mghm","updated_at":"2019-05-03T17:52:16Z","created_at":"2019-05-03T17:52:16Z","number":2,"locked":false,"comment":"","active":false,"deployed":false,"testing":false},{"deployed":false,"active":false,"comment":"","locked":false,"number":3,"testing":false,"service_id":"5stIxTZWQz7D8B1Am1Mghm","staging":false,"deleted_at":null,"updated_at":"2019-05-03T17:53:01Z","created_at":"2019-05-03T17:53:01Z"},{"testing":false,"comment":"","number":4,"locked":false,"deployed":false,"active":false,"created_at":"2019-05-03T18:01:46Z","updated_at":"2019-05-03T18:01:46Z","deleted_at":null,"service_id":"5stIxTZWQz7D8B1Am1Mghm","staging":false}],"comment":"Managed
+      comment","active":false,"deployed":false},{"deleted_at":null,"service_id":"7i6HN3TK9wS159v2gPAZ8A","staging":false,"updated_at":"2019-05-11T00:21:36Z","created_at":"2019-05-11T00:21:36Z","comment":"","number":50,"locked":false,"active":false,"deployed":false,"testing":false},{"testing":false,"active":false,"deployed":false,"comment":"","locked":false,"number":51,"updated_at":"2019-05-11T00:32:55Z","created_at":"2019-05-11T00:32:53Z","service_id":"7i6HN3TK9wS159v2gPAZ8A","staging":false,"deleted_at":null}],"customer_id":"76eo17QE8I25lqXy9uJqbd"},{"comment":"comment","id":"7i6HN3TK9wS159v2gPAZ8A","versions":[{"created_at":"2019-05-11T17:03:47Z","updated_at":"2019-05-11T17:03:47Z","deleted_at":null,"staging":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","testing":false,"number":1,"locked":false,"comment":"","active":false,"deployed":false}],"customer_id":"76eo17QE8I25lqXy9uJqbd","created_at":"2019-05-11T17:03:47Z","updated_at":"2019-05-11T17:03:47Z","name":"test-service","version":1},{"id":"5stIxTZWQz7D8B1Am1Mghm","versions":[{"updated_at":"2019-05-03T16:02:42Z","created_at":"2019-05-03T16:02:37Z","deleted_at":null,"staging":false,"service_id":"5stIxTZWQz7D8B1Am1Mghm","testing":false,"locked":true,"number":1,"comment":"","active":true,"deployed":false},{"deleted_at":null,"staging":false,"service_id":"5stIxTZWQz7D8B1Am1Mghm","updated_at":"2019-05-03T17:52:16Z","created_at":"2019-05-03T17:52:16Z","number":2,"locked":false,"comment":"","active":false,"deployed":false,"testing":false},{"deployed":false,"active":false,"comment":"","locked":false,"number":3,"testing":false,"service_id":"5stIxTZWQz7D8B1Am1Mghm","staging":false,"deleted_at":null,"updated_at":"2019-05-03T17:53:01Z","created_at":"2019-05-03T17:53:01Z"},{"testing":false,"comment":"","number":4,"locked":false,"deployed":false,"active":false,"created_at":"2019-05-03T18:01:46Z","updated_at":"2019-05-03T18:01:46Z","deleted_at":null,"service_id":"5stIxTZWQz7D8B1Am1Mghm","staging":false}],"comment":"Managed
       by Terraform","customer_id":"76eo17QE8I25lqXy9uJqbd","updated_at":"2019-05-03T16:02:37Z","created_at":"2019-05-03T16:02:37Z","name":"wework","version":1}]'
     headers:
       Accept-Ranges:
diff --git a/fastly/fixtures/services/search.yaml b/fastly/fixtures/services/search.yaml
index 247a450..f7d0743 100644
--- a/fastly/fixtures/services/search.yaml
+++ b/fastly/fixtures/services/search.yaml
@@ -13,7 +13,7 @@ interactions:
     url: https://api.fastly.com/service/search?name=test-service
     method: GET
   response:
-    body: '{"deleted_at":null,"name":"test-service","updated_at":"2019-05-11T17:03:47Z","created_at":"2019-05-11T17:03:47Z","customer_id":"76eo17QE8I25lqXy9uJqbd","id":"4cyy8qad5xykKfjyTCkGnq","versions":[{"testing":false,"active":false,"deployed":false,"comment":"","locked":false,"number":1,"updated_at":"2019-05-11T17:03:47Z","created_at":"2019-05-11T17:03:47Z","service_id":"4cyy8qad5xykKfjyTCkGnq","staging":false,"deleted_at":null}],"comment":"comment"}'
+    body: '{"deleted_at":null,"name":"test-service","updated_at":"2019-05-11T17:03:47Z","created_at":"2019-05-11T17:03:47Z","customer_id":"76eo17QE8I25lqXy9uJqbd","id":"7i6HN3TK9wS159v2gPAZ8A","versions":[{"testing":false,"active":false,"deployed":false,"comment":"","locked":false,"number":1,"updated_at":"2019-05-11T17:03:47Z","created_at":"2019-05-11T17:03:47Z","service_id":"7i6HN3TK9wS159v2gPAZ8A","staging":false,"deleted_at":null}],"comment":"comment"}'
     headers:
       Accept-Ranges:
       - bytes
diff --git a/fastly/fixtures/services/update.yaml b/fastly/fixtures/services/update.yaml
index 0d5605e..743816b 100644
--- a/fastly/fixtures/services/update.yaml
+++ b/fastly/fixtures/services/update.yaml
@@ -3,10 +3,10 @@ version: 1
 rwmutex: {}
 interactions:
 - request:
-    body: ID=4cyy8qad5xykKfjyTCkGnq&name=new-test-service
+    body: ID=7i6HN3TK9wS159v2gPAZ8A&name=new-test-service
     form:
       ID:
-      - 4cyy8qad5xykKfjyTCkGnq
+      - 7i6HN3TK9wS159v2gPAZ8A
       name:
       - new-test-service
     headers:
@@ -16,10 +16,10 @@ interactions:
       - xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
       User-Agent:
       - FastlyGo/0.4.3.dev (+github.com/fastly/go-fastly; go1.12)
-    url: https://api.fastly.com/service/4cyy8qad5xykKfjyTCkGnq
+    url: https://api.fastly.com/service/7i6HN3TK9wS159v2gPAZ8A
     method: PUT
   response:
-    body: '{"name":"new-test-service","customer_id":"76eo17QE8I25lqXy9uJqbd","publish_key":"ff100f0c03c61271b4052b07878159c721c4e897","created_at":"2019-05-11T17:03:47Z","comment":"comment","updated_at":"2019-05-11T17:03:47Z","id":"4cyy8qad5xykKfjyTCkGnq","versions":[{"created_at":"2019-05-11T17:03:47Z","locked":false,"active":false,"staging":false,"deleted_at":null,"testing":false,"comment":"","service_id":"4cyy8qad5xykKfjyTCkGnq","number":1,"deployed":false,"updated_at":"2019-05-11T17:03:47Z"}],"deleted_at":null}'
+    body: '{"name":"new-test-service","customer_id":"76eo17QE8I25lqXy9uJqbd","publish_key":"ff100f0c03c61271b4052b07878159c721c4e897","created_at":"2019-05-11T17:03:47Z","comment":"comment","updated_at":"2019-05-11T17:03:47Z","id":"7i6HN3TK9wS159v2gPAZ8A","versions":[{"created_at":"2019-05-11T17:03:47Z","locked":false,"active":false,"staging":false,"deleted_at":null,"testing":false,"comment":"","service_id":"7i6HN3TK9wS159v2gPAZ8A","number":1,"deployed":false,"updated_at":"2019-05-11T17:03:47Z"}],"deleted_at":null}'
     headers:
       Accept-Ranges:
       - bytes
diff --git a/fastly/service_test.go b/fastly/service_test.go
index 6224bb2..3b7bb21 100644
--- a/fastly/service_test.go
+++ b/fastly/service_test.go
@@ -101,7 +101,7 @@ func TestClient_Services(t *testing.T) {
 		t.Errorf("Service Detail Version is empty: (%#v)", nsd)
 	}
 	if nsd.ActiveVersion.Number == 0 {
-		t.Errorf("Service Active Version is empty: (%#v)", nsd)
+		t.Errorf("Service Active Version is empty: (%#v)", nsd.ActiveVersion)
 	}
 
 	// Search
diff --git a/fastly/version.go b/fastly/version.go
index 5fbf7f1..407b0f4 100644
--- a/fastly/version.go
+++ b/fastly/version.go
@@ -22,12 +22,19 @@ type Version struct {
 }
 
 type ActiveVersion struct {
-	Version
-
+	Number         int                    `mapstructure:"number"`
+	Comment        string                 `mapstructure:"comment"`
+	ServiceID      string                 `mapstructure:"service_id"`
+	Active         bool                   `mapstructure:"active"`
+	Locked         bool                   `mapstructure:"locked"`
+	Deployed       bool                   `mapstructure:"deployed"`
+	Staging        bool                   `mapstructure:"staging"`
+	Testing        bool                   `mapstructure:"testing"`
 	Backends       []Backend              `mapstructure:"backends"`
 	CacheSetting   []CacheSetting         `mapstructure:"cache_settings"`
 	Conditions     []Condition            `mapstructure:"conditions"`
 	Dictionaries   []Dictionary           `mapstructure:"dictionaries"`
+	Directors      []Director             `mapstructure:"directors"`
 	Domains        []Domain               `mapstructure:"domains"`
 	Gzips          []Gzip                 `mapstructure:"gzips"`
 	Headers        []Header               `mapstructure:"headers"`
```

ovv commented:
Perfect thank you. We should be good to :shipit: now.

> There's probably a way to solve that (and if you're aware of how to do that then great) but it didn't seem necessary to go down a rabbit hole for the sake of a few extra lines of code

I couldn't find a straightforward way to do that. If you are fine with a bit of duplicated line I think we can leave it like that.

Integralist commented on fastly/fixtures/services/details.yaml:
```diff
@@ -1,46 +1,44 @@
     method: GET
   response:
-    body: '{"name":"test-service","created_at":"2019-05-11T17:03:47Z","updated_at":"2019-05-11T17:03:47Z","deleted_at":null,"versions":[{"created_at":"2019-05-11T17:03:47Z","locked":false,"active":false,"staging":false,"deleted_at":null,"testing":false,"comment":"","service_id":"4cyy8qad5xykKfjyTCkGnq","number":1,"deployed":false,"updated_at":"2019-05-11T17:03:47Z"}],"id":"4cyy8qad5xykKfjyTCkGnq" ...
+    body: '{"created_at":"2021-01-13T18:56:56Z","type":"vcl","versions":[{"created_at":"2021-01-13T18:56:56Z","testing":false,"deployed":false,"deleted_at":null,"staging":false,"comment":"","active":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","updated_at":"2021-01-13T18:56:56Z","number":1,"locked":false},{"active":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","deleted_at":null,"comment":"","staging ...
```
@ovv would you mind replacing `integralist-test-go-fastly-tests.com` in the below fixture output with `test-service.com`. It's not a concern it being there, but would be better if my username wasn't in the fixture data really 🙂 

This shouldn't break any tests as nothing is validating the domain AFAIK.

Thanks!

Integralist reviewed (changes requested):
@ovv Just a quick change please.

Integralist commented on fastly/fixtures/services/details.yaml:
```diff
@@ -1,46 +1,44 @@
+    body: '{"created_at":"2021-01-13T18:56:56Z","type":"vcl","versions":[{"created_at":"2021-01-13T18:56:56Z","testing":false,"deployed":false,"deleted_at":null,"staging":false,"comment":"","active":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","updated_at":"2021-01-13T18:56:56Z","number":1,"locked":false},{"active":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","deleted_at":null,"comment":"","staging ...
+      comment","staging":false,"deleted_at":null,"active":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","created_at":"2021-01-14T09:51:27Z","testing":false,"locked":true,"updated_at":"2021-01-14T09:51:30Z","number":17},{"locked":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","updated_at":"2021-01-14T09:51:30Z","number":18,"deployed":false,"comment":"","staging":false,"deleted_at":null,"active":false," ...
+      1","ssl_sni_hostname":"httpbin.org","ssl_cert_hostname":"httpbin.org","between_bytes_timeout":10000,"hostname":"httpbin.org","ssl_client_key":null,"ssl_check_cert":true,"port":443,"ipv4":null,"comment":"","ssl_client_cert":null,"healthcheck":null,"ssl_hostname":null,"client_cert":null,"first_byte_timeout":15000,"ssl_ciphers":null,"request_condition":"","auto_loadbalance":false,"min_tls_vers ...
+      1","ssl_sni_hostname":"httpbin.org","ssl_cert_hostname":"httpbin.org","between_bytes_timeout":10000,"hostname":"httpbin.org","ssl_client_key":null,"ssl_check_cert":true,"port":443,"ipv4":null,"comment":"","ssl_client_cert":null,"healthcheck":null,"ssl_hostname":null,"client_cert":null,"first_byte_timeout":15000,"ssl_ciphers":null,"request_condition":"","auto_loadbalance":false,"min_tls_vers ...
```
```suggestion
      1","ssl_sni_hostname":"httpbin.org","ssl_cert_hostname":"httpbin.org","between_bytes_timeout":10000,"hostname":"httpbin.org","ssl_client_key":null,"ssl_check_cert":true,"port":443,"ipv4":null,"comment":"","ssl_client_cert":null,"healthcheck":null,"ssl_hostname":null,"client_cert":null,"first_byte_timeout":15000,"ssl_ciphers":null,"request_condition":"","auto_loadbalance":false,"min_tls_version":null,"shield":null,"max_tls_version":null,"error_threshold":0,"connect_timeout":1000}],"cache_settings":[],"conditions":[],"dictionaries":[],"directors":[],"domains":[{"name":"test-service.com","comment":"comment"}],"gzips":[],"headers":[],"healthchecks":[],"request_settings":[],"response_objects":[],"snippets":[],"vcls":[],"wordpress":[],"settings":{"general.stale_if_error":false,"general.default_pci":0,"general.default_host":"","general.default_ttl":3600,"general.stale_if_error_ttl":43200}}}'
```

Integralist commented on fastly/fixtures/services/details.yaml:
```diff
@@ -1,46 +1,44 @@
-    body: '{"name":"test-service","created_at":"2019-05-11T17:03:47Z","updated_at":"2019-05-11T17:03:47Z","deleted_at":null,"versions":[{"created_at":"2019-05-11T17:03:47Z","locked":false,"active":false,"staging":false,"deleted_at":null,"testing":false,"comment":"","service_id":"4cyy8qad5xykKfjyTCkGnq","number":1,"deployed":false,"updated_at":"2019-05-11T17:03:47Z"}],"id":"4cyy8qad5xykKfjyTCkGnq" ...
+    body: '{"created_at":"2021-01-13T18:56:56Z","type":"vcl","versions":[{"created_at":"2021-01-13T18:56:56Z","testing":false,"deployed":false,"deleted_at":null,"staging":false,"comment":"","active":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","updated_at":"2021-01-13T18:56:56Z","number":1,"locked":false},{"active":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","deleted_at":null,"comment":"","staging ...
+      comment","staging":false,"deleted_at":null,"active":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","created_at":"2021-01-14T09:51:27Z","testing":false,"locked":true,"updated_at":"2021-01-14T09:51:30Z","number":17},{"locked":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","updated_at":"2021-01-14T09:51:30Z","number":18,"deployed":false,"comment":"","staging":false,"deleted_at":null,"active":false," ...
+      1","ssl_sni_hostname":"httpbin.org","ssl_cert_hostname":"httpbin.org","between_bytes_timeout":10000,"hostname":"httpbin.org","ssl_client_key":null,"ssl_check_cert":true,"port":443,"ipv4":null,"comment":"","ssl_client_cert":null,"healthcheck":null,"ssl_hostname":null,"client_cert":null,"first_byte_timeout":15000,"ssl_ciphers":null,"request_condition":"","auto_loadbalance":false,"min_tls_vers ...
```
```suggestion
      1","ssl_sni_hostname":"httpbin.org","ssl_cert_hostname":"httpbin.org","between_bytes_timeout":10000,"hostname":"httpbin.org","ssl_client_key":null,"ssl_check_cert":true,"port":443,"ipv4":null,"comment":"","ssl_client_cert":null,"healthcheck":null,"ssl_hostname":null,"client_cert":null,"first_byte_timeout":15000,"ssl_ciphers":null,"request_condition":"","auto_loadbalance":false,"min_tls_version":null,"shield":null,"max_tls_version":null,"error_threshold":0,"connect_timeout":1000}],"cache_settings":[],"conditions":[],"dictionaries":[],"directors":[],"domains":[{"name":"test-service.com","comment":"comment"}],"gzips":[],"headers":[],"healthchecks":[],"request_settings":[],"response_objects":[],"snippets":[],"vcls":[],"wordpress":[],"settings":{"general.stale_if_error":false,"general.default_pci":0,"general.default_host":"","general.default_ttl":3600,"general.stale_if_error_ttl":43200}},"active_version":{"locked":true,"created_at":"2021-01-14T14:22:42Z","active":true,"updated_at":"2021-01-14T14:36:25Z","testing":false,"deleted_at":null,"deployed":false,"number":71,"comment":"","staging":false,"service_id":"7i6HN3TK9wS159v2gPAZ8A","acls":[],"backends":[{"ssl_ca_cert":null,"use_ssl":true,"ipv6":null,"weight":100,"override_host":null,"max_conn":200,"address":"httpbin.org","name":"Host
```

ovv commented:
> @ovv Just a quick change please.

I think I got them all. Let me know if there is anything else needed

Integralist commented on fastly/fixtures/services/create.yaml:
```diff
@@ -19,7 +19,7 @@ interactions:
     method: POST
   response:
-    body: '{"customer_id":"76eo17QE8I25lqXy9uJqbd","comment":"comment","name":"test-service","created_at":"2019-05-11T17:03:47Z","publish_key":"ff100f0c03c61271b4052b07878159c721c4e897","deleted_at":null,"versions":[{"staging":false,"updated_at":"2019-05-11T17:03:47Z","deployed":false,"comment":"","service_id":"4cyy8qad5xykKfjyTCkGnq","locked":false,"number":1,"deleted_at":null,"testing":false,"c ...
+    body: '{"customer_id":"76eo17QE8I25lqXy9uJqbd","comment":"comment","name":"test-service","created_at":"2019-05-11T17:03:47Z","publish_key":"ff100f0c03c61271b4052b07878159c721c4e897","deleted_at":null,"versions":[{"staging":false,"updated_at":"2019-05-11T17:03:47Z","deployed":false,"comment":"","service_id":"7i6HN3TK9wS159v2gPAZ8A","locked":false,"number":1,"deleted_at":null,"testing":false,"c ...
```
Hmm, just noticed something isn't quite right with this service fixture. It's missing `paused` and `type` (and if I compare it to another PR where I updated the fixture for the service I can see those fields are included) 🤔

Integralist replied:
I've also noticed the Go-Fastly User-Agent hasn't been updated on these recreated fixtures, which is interesting. 

So what I'll likely do is get my other PR merged first (which is updating all fixture files) and then I might have to come back to this PR due to some confusion I have around what's documented in the official API reference.

Specifically, if I look at the `/service/service_id/details` endpoint I can see they document both a `version` and a `active_version` key and _both_ contain the same fields so it's unclear to me at this point if that's an error on the API side or an intentional decision.

@ovv with this in mind ^^ I'll have to hold off merging this PR until I can get some clarity.

ovv replied:
Sure :+1: 

I just made sure the "allow edit from maintainers" is ticked. If you need to do some changes to get this merged feel free to go ahead. It doesn't sounds like I'm needed to drive this forward and you have much more expertise in the codebase.

Thanks for your time, have a great week.

Integralist reviewed (commented):
Hi @ovv 

I'm going to put this PR on hold for the moment (see my comments below for details). I'll come back to this once I have some questions answered from the relevant API team internally. 

Thanks.

Integralist commented:
Just wanted to circle back to this PR to say that I had a response from the API team, which clarifies the difference between the `active_version` and `version` fields...

> The `active_version` is always the active version or nil if no version is active.
>
> The `version` is a bit more subtle. 
> 
> If you pass a version param to the GET request it will be that version. 
> 
> Otherwise it defaults to the most recent version.
>
> So if the most recent version is the active version and you haven’t passed a version param to the GET, they will be the same.

At this point we've already proceeded with the Go-Fastly 3.0.0 release, and as this PR introduces a breaking change it would have to wait until the next major release version (which would be 4.0.0).

For now I'm going to close this PR and circle back around to it whenever the time comes for the next major release.
